docs/api-reference/public_api.rst

```diff
@@ -0,0 +1,12 @@
+:orphan:
```

Comment:
Shouldn't we just add a toctree to `api-reference/index.rst` somewhere rather than marking this as an orphan page?

Reply:
Sure, no objections. I just used the same scheme as for `docs/porting-example/steps/step_*.rst`.

Reply:
The step files were only removed because I didn't want them to distract people from the text of the porting guide or clutter the side bar menu. I think the core API is worthy of being in a toctree though.